nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1321,22 +1321,7 @@ impl DataStore {
         .await
     }
 
-    /// Delete the provided `authz_instance`, as long as it is in one of the
-    /// provided set of [`InstanceState`]s.
-    ///
-    /// A.K.A. "[`project_delete_instance`], hard mode". Typically, instances
-    /// may only be deleted if they are in the [`InstanceState::NoVmm`] or
-    /// `[`InstanceState::Failed`] states. This method exists separately in
-    /// order to allow an unwinding `instance-create` saga to delete the
-    /// instance record without transiently moving it to
-    /// [`InstanceState::Failed`], in which it becomes eligible for automatic
-    /// restart.
-    ///
-    /// In general, callers should use [`project_delete_instance`] instead,
-    /// unless you really know what you're doing.
-    ///
-    /// [`project_delete_instance`]: Self::project_delete_instance
-    pub async fn project_delete_instance_in_states(
+    async fn project_idempotently_mark_instance_deleted_in_states(
```

Comment:
Nitpicky, feel free to ignore: I don't _love_ this name: `mark_instance_deleted` kinda implies to me that this _just_ changes the state to `Destroyed` and doesn't do any of the other work of the deletion. But, looking at the code, this seems to also detach resources. I'm not sure if I can think of a better name for it off the top of my head, though...